### PR TITLE
Bug 1135044 - When updating a call group's contact information dispatch the appropriate udpate event r=drs

### PR DESCRIPTION
**Pull request by gabrielesvelto**

### CODE REVIEW

File: apps/communications/dialer/js/call_log.js [hunk 1]
```diff
@@ -1077,19 +1072,10 @@ var CallLog = {
...
-    if (contact.tel && contact.tel.length) {
-      phoneNumbers = contact.tel.map(function(tel) {
-        return tel.value;
-      });
-    }
     switch (reason) {
-      case 'create':
-        CallLog.updateListWithContactInfo('create', null, phoneNumbers);
-        break;
+      case 'create': /* FALLTHROUGH */
```
Comment (DouglasSherk) [inline apps/communications/dialer/js/call_log.js]: This really doesn't have to be a ```switch``` anymore.

File: apps/communications/dialer/js/call_log_db.js [hunk 2]
```diff
@@ -1305,6 +1305,7 @@ var CallLogDBManager = {
             }
           }
           cursor.update(group);
+          self._dispatchCallLogDbNewCall(self._getGroupObject(group));
```
Comment (DouglasSherk) [inline apps/communications/dialer/js/call_log_db.js]: This needs a test.

File: apps/communications/dialer/js/call_log.js [hunk 3]
```diff
@@ -943,16 +943,11 @@ var CallLog = {
...
-   *        Array of phoneNumbers associated with a contact. Only
-   *        'oncontactchange' events with 'update' or 'add' reasons will
-   *        provide this paramater.
    * param target
    *        DOM element to be updated. We default to the whole log if no
    *        'target' param is provided.
    */
-  updateListWithContactInfo: function cl_updateList(reason, contactId,
-                                                    phoneNumbers, target) {
+  updateListWithContactInfo: function cl_updateList(reason, contactId, target) {
```
Comment (DouglasSherk) [inline apps/communications/dialer/js/call_log.js]: I'd prefer if this had tests, but since it's just a cleanup, I'm not as concerned about it.

**Comment by try-server-hook** [comment]
[gabrielesvelto Gabriele Svelto (gabrielesvelto)](https://github.com/gabrielesvelto) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=da71cdc6a601)

**Comment by try-server-hook** [comment]
[gabrielesvelto Gabriele Svelto (gabrielesvelto)](https://github.com/gabrielesvelto) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=4fe386b3fc6a)

File: apps/communications/dialer/js/call_log.js [hunk 4]
```diff
@@ -1085,21 +1080,7 @@ var CallLog = {
 
 navigator.mozContacts.oncontactchange = function oncontactchange(event) {
   function contactChanged(contact, reason) {
```
Comment (DouglasSherk) [inline apps/communications/dialer/js/call_log.js]: We don't really need this function anymore. Let's just inline ```CallLog.updateListWithContactInfo(reason, event.contactID);```.

**Comment by try-server-hook** [comment]
[gabrielesvelto Gabriele Svelto (gabrielesvelto)](https://github.com/gabrielesvelto) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=47027cb53787)

**Pull request merged**